src/lab.rs

```diff
@@ -224,69 +225,75 @@ impl Div<f32> for Lab {
 
 from_color!(to Lab from Rgb, Luma, Xyz, Lch, Hsv, Hsl);
 
-impl From<Xyz> for Lab {
-    fn from(xyz: Xyz) -> Lab {
+impl<T: Float> From<Xyz<T>> for Lab<T> {
+    fn from(xyz: Xyz<T>) -> Lab<T> {
         Lab {
-            l: (116.0 * f(xyz.y / Y_N) - 16.0) / 100.0,
-            a: (500.0 * (f(xyz.x / X_N) - f(xyz.y / Y_N))) / 128.0,
-            b: (200.0 * (f(xyz.y / Y_N) - f(xyz.z / Z_N))) / 128.0,
+            l: (T::from(116.0).unwrap() * f(xyz.y / T::from(Y_N).unwrap()) -
+                T::from(16.0).unwrap()) / T::from(100.0).unwrap(),
+            a: (T::from(500.0).unwrap() *
+                (f(xyz.x / T::from(X_N).unwrap()) - f(xyz.y / T::from(Y_N).unwrap()))) /
+               T::from(128.0).unwrap(),
+            b: (T::from(200.0).unwrap() *
+                (f(xyz.y / T::from(Y_N).unwrap()) - f(xyz.z / T::from(Z_N).unwrap()))) /
+               T::from(128.0).unwrap(),
             alpha: xyz.alpha,
         }
     }
 }
 
-impl From<Rgb> for Lab {
-    fn from(rgb: Rgb) -> Lab {
+impl<T: Float> From<Rgb<T>> for Lab<T> {
+    fn from(rgb: Rgb<T>) -> Lab<T> {
         Xyz::from(rgb).into()
     }
 }
 
-impl From<Luma> for Lab {
-    fn from(luma: Luma) -> Lab {
+impl<T: Float> From<Luma<T>> for Lab<T> {
+    fn from(luma: Luma<T>) -> Lab<T> {
         Xyz::from(luma).into()
     }
 }
 
-impl From<Lch> for Lab {
-    fn from(lch: Lch) -> Lab {
+impl<T: Float> From<Lch<T>> for Lab<T> {
+    fn from(lch: Lch<T>) -> Lab<T> {
         Lab {
             l: lch.l,
-            a: lch.chroma.max(0.0) * lch.hue.to_radians().cos(),
-            b: lch.chroma.max(0.0) * lch.hue.to_radians().sin(),
+            a: lch.chroma.max(T::zero()) * lch.hue.to_radians().cos(),
+            b: lch.chroma.max(T::zero()) * lch.hue.to_radians().sin(),
             alpha: lch.alpha,
         }
     }
 }
 
-impl From<Hsv> for Lab {
-    fn from(hsv: Hsv) -> Lab {
+impl<T: Float> From<Hsv<T>> for Lab<T> {
+    fn from(hsv: Hsv<T>) -> Lab<T> {
         Xyz::from(hsv).into()
     }
 }
 
-impl From<Hsl> for Lab {
-    fn from(hsl: Hsl) -> Lab {
+impl<T: Float> From<Hsl<T>> for Lab<T> {
+    fn from(hsl: Hsl<T>) -> Lab<T> {
         Xyz::from(hsl).into()
     }
 }
 
-fn f(t: f32) -> f32 {
-    //(6/29)^3
-    const C_6_O_29_P_3: f32 = 0.00885645167;
-    //(29/6)^2
-    const C_29_O_6_P_2: f32 = 23.3611111111;
+fn f<T: Float>(t: T) -> T {
+    // (T::from(6/29).unwrap())^3
```

Comment:
No need to change these comments. They are just there to show where the numbers comes from.